### PR TITLE
Bugfix: font sizes in payment-buttons taking precedence over contained buttons

### LINKED ISSUE #26748: Payment buttons: inconsistent button size between editor and front-end
### Impacted plugin

Jetpack

### Steps to Reproduce

1. Add 2 payment buttons.
2. Update the size in the right setting panel. Make one Extra small and one Normal.
3. Select the payment buttons block and change the size to Extra small. 
4. Save page and check it in frontend.

### A clear and concise description of what you expected to happen.

Buttons have the selected sizes. One extra small and one normal.

### What actually happened

Buttons have the same size in the front end, but different size (as expected) in the editor.

### Browser

Google Chrome/Chromium

### Other information

editor
frontend

### Platform (Simple, Atomic, or both?)

_No response_

### Reproducibility

Consistent

### Severity

_No response_

### Available workarounds?

_No response_

### Workaround details

_No response_

### PR DESCRIPTION
**Pull request by dsas**
Fixes #26748

#### Changes proposed in this Pull Request:

The payment-buttons typography feature introduced in #26580 had a bug
which meant that the font size in the payment-buttons block was
overriding font sizes set on the buttons when a preset font size was
being used. This bug occured when viewing the site, but not when
editing.
    
The bug was caused by the view CSS selector not being quite specific
enough.
    
The font size of the button blocks when using a preset works by adding a
has-{presetName}-font-size class to the element in question, where
presetName varies depending upon the theme. To fix this bug whenever a
preset font size is used a class called has-custom-font-size is added.
This gives a known stable class name which is targeted with CSS.

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

 1. Apply patch / add a JN site etc
 2. Open Gutenberg
 3. Add 4 payment buttons.
 4. Update the size in the right setting panel. Make one small, one XL, one a set font size e.g. 15px and don't give any font-size to the last
 5. Select the payment buttons block and change the size to small 
 6. Save page
 7. Check it in frontend, it should look like in the editor: two small buttons at either end and in the middle an XL button and small looking one with a font-size of 15px

### CODE REVIEW

**Review by mmtr — DISMISSED** [review]
This was fun to test, until I realized that the issue was theme-specific 😄 . It does work well, although it doesn't autofix existing buttons – they need to be re-created, but that's ok I think!

**Comment by dsas** [comment]
> although it doesn't autofix existing buttons – they need to be re-created, but that's ok I think!

I was worried about that but couldn't figure out a better way given the names of the font size classes are theme specific. We've not actually had a jetpack release since then, so it's just some wpcom users adding buttons with typography settings the last week who will need to re-save to get things displayed perfectly.

File: projects/plugins/jetpack/extensions/blocks/button/button.php [hunk 1]
```diff
@@ -109,6 +109,7 @@ function get_button_classes( $attributes ) {
 
 	if ( $has_font_size ) {
 		$classes[] = 'has-' . $attributes['fontSize'] . '-font-size';
+		$classes[] = 'has-custom-font-size';
```
Comment (dsas) [inline projects/plugins/jetpack/extensions/blocks/button/button.php:112]: I am simultaneously sure this pr worked before this change, but also know that it couldn't have.

The jetpack/button will only save the `has-custom-font-size` class added in save.js if the `saveInPostContent` attribute is true. That attribute is never true for payment-buttons and so it relies upon this php code instead.

Comment (mmtr) [reply projects/plugins/jetpack/extensions/blocks/button/button.php:112]: > I am simultaneously sure this pr worked before this change, but also know that it couldn't have.

The Schrödinger's class!

**Review by mmtr — APPROVED** [review]

**Pull request merged**